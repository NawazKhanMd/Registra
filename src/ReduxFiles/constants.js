export const actions = {
    loading: 'loading',
    stage: 'stage',
    clear_store: 'clear_store',
    save_user_data: 'save_user_data',
    profile_picture: 'profile_picture',
    user_sign: 'user_sign',
    save_office_data: 'save_office_data',
    save_user_data_api: 'save_user_data_api',
    save_office_data_api: 'save_office_data_api',
    save_office_data_success: 'save_office_data_success',
    save_office_data_failure: 'save_office_data_failure',
    save_user_data_success: 'save_user_data_success',
    save_user_data_failure: 'save_user_data_failure',
} 