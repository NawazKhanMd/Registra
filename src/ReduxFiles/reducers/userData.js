import { actions } from "../constants";

const EmptyUserInfo = {
    "usr_name": "",
    "usr_phn": "",
    "usr_email": "",
    "usr_add_1": "",
    "usr_add_2": "",
    "usr_add_3": ""
}
const EmptyOfficeData = {
    "building_name": "",
    "city_area": "",
    "landline_number": "",
    "building_add_1": "",
    "building_add_2": "",
    "building_add_3": ""
}

export const initialState = {
    userInfo: EmptyUserInfo, officeData: EmptyOfficeData, profilePicture: "", userSignature: ""
};


export const UserData = (state = initialState, action) => {
    switch (action.type) {
        case actions.save_user_data:
            return { ...state, userInfo: { ...action.payload } }
        case actions.save_office_data:
            return { ...state, officeData: { ...action.payload } }
        case actions.clear_store:
            return {  ...state, userInfo: {...EmptyUserInfo}, officeData: {...EmptyOfficeData} }
        case actions.profile_picture:
            return { ...state, profilePicture: action.payload }
        case actions.user_sign:
            return { ...state, userSignature: action.payload }
        default:
            return state
    }
}