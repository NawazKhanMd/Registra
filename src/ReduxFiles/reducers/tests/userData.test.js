import { UserData, initialState } from '../userData';
import { actions } from '../../constants';
//import { dummyVariousContent } from '../types/contentTypes';

describe('Home Reducer', () => {
    it('Handles default case', () => {
        const state = UserData(undefined, {});
        expect(state).toEqual(initialState);
    });
    
    it('Handles clear Store ', () => {
        const action = { type: actions.clear_store};
        const state = UserData(undefined, action);
        expect(state).toMatchSnapshot();
    });
    it('Handles load User Info ', () => {
        const data = {
            "usr_name": "Md Nawaz Khan",
            "usr_phn": "9121223331",
            "usr_email": "nawazmd230@gmail.com",
            "usr_add_1": "Mukti",
            "usr_add_2": "Malleshwaram",
            "usr_add_3": "IN, 560003"
        }
        const action = { type: actions.save_user_data, payload: data };
        const state = UserData(undefined, action);
        expect(state).toMatchSnapshot();
    });
    it('Handles load Office Info ', () => {
        const data = {
            "building_name": "Mukti",
            "city_area": "Malleshwaram",
            "landline_number": "0802345225",
            "building_add_1": "Mukti",
            "building_add_2": "Malleshwaram",
            "building_add_3": "IN, 560003"
        }
        const action = { type: actions.save_office_data, payload: data };
        const state = UserData(undefined, action);
        expect(state).toMatchSnapshot();
    });
    it('Handles profile picture save ', () => {
        const data =  "Profile_picture";
        const action = { type: actions.profile_picture, payload: data };
        const state = UserData(undefined, action);
        expect(state).toMatchSnapshot();
    });
    it('Handles Signature save ', () => {
        const data =  "user_signature";
        const action = { type: actions.user_sign, payload: data };
        const state = UserData(undefined, action);
        expect(state).toMatchSnapshot();
    });
});