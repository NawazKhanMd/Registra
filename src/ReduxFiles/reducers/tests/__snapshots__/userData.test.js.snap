// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Reducer Handles Signature save  1`] = `
Object {
  "officeData": Object {
    "building_add_1": "",
    "building_add_2": "",
    "building_add_3": "",
    "building_name": "",
    "city_area": "",
    "landline_number": "",
  },
  "profilePicture": "",
  "userInfo": Object {
    "usr_add_1": "",
    "usr_add_2": "",
    "usr_add_3": "",
    "usr_email": "",
    "usr_name": "",
    "usr_phn": "",
  },
  "userSignature": "user_signature",
}
`;

exports[`Home Reducer Handles clear Store  1`] = `
Object {
  "officeData": Object {
    "building_add_1": "",
    "building_add_2": "",
    "building_add_3": "",
    "building_name": "",
    "city_area": "",
    "landline_number": "",
  },
  "profilePicture": "",
  "userInfo": Object {
    "usr_add_1": "",
    "usr_add_2": "",
    "usr_add_3": "",
    "usr_email": "",
    "usr_name": "",
    "usr_phn": "",
  },
  "userSignature": "",
}
`;

exports[`Home Reducer Handles load 1 movie  1`] = `
Object {
  "officeData": Object {
    "building_add_1": "",
    "building_add_2": "",
    "building_add_3": "",
    "building_name": "",
    "city_area": "",
    "landline_number": "",
  },
  "profilePicture": "",
  "userInfo": Object {
    "usr_add_1": "Mukti",
    "usr_add_2": "Malleshwaram",
    "usr_add_3": "IN, 560003",
    "usr_email": "nawazmd230@gmail.com",
    "usr_name": "Md Nawaz Khan",
    "usr_phn": "9121223331",
  },
  "userSignature": "",
}
`;

exports[`Home Reducer Handles load Office Info  1`] = `
Object {
  "officeData": Object {
    "building_add_1": "Mukti",
    "building_add_2": "Malleshwaram",
    "building_add_3": "IN, 560003",
    "building_name": "Mukti",
    "city_area": "Malleshwaram",
    "landline_number": "0802345225",
  },
  "profilePicture": "",
  "userInfo": Object {
    "usr_add_1": "",
    "usr_add_2": "",
    "usr_add_3": "",
    "usr_email": "",
    "usr_name": "",
    "usr_phn": "",
  },
  "userSignature": "",
}
`;

exports[`Home Reducer Handles load User Info  1`] = `
Object {
  "officeData": Object {
    "building_add_1": "",
    "building_add_2": "",
    "building_add_3": "",
    "building_name": "",
    "city_area": "",
    "landline_number": "",
  },
  "profilePicture": "",
  "userInfo": Object {
    "usr_add_1": "Mukti",
    "usr_add_2": "Malleshwaram",
    "usr_add_3": "IN, 560003",
    "usr_email": "nawazmd230@gmail.com",
    "usr_name": "Md Nawaz Khan",
    "usr_phn": "9121223331",
  },
  "userSignature": "",
}
`;

exports[`Home Reducer Handles profile picture save  1`] = `
Object {
  "officeData": Object {
    "building_add_1": "",
    "building_add_2": "",
    "building_add_3": "",
    "building_name": "",
    "city_area": "",
    "landline_number": "",
  },
  "profilePicture": "Profile_picture",
  "userInfo": Object {
    "usr_add_1": "",
    "usr_add_2": "",
    "usr_add_3": "",
    "usr_email": "",
    "usr_name": "",
    "usr_phn": "",
  },
  "userSignature": "",
}
`;
